modules/light-clients/08-wasm/types/upgrade_test.go

```diff
@@ -98,536 +106,145 @@ func (suite *TypesTestSuite) TestVerifyUpgradeGrandpa() {
+					var payload types.SudoMsg
+
+					err := json.Unmarshal(sudoMsg, &payload)
+					suite.Require().NoError(err)
+
+					// verify payload values
+					suite.Require().Equal(upgradedClient, &payload.VerifyUpgradeAndUpdateState.UpgradeClientState)
+					suite.Require().Equal(upgradedConsState, &payload.VerifyUpgradeAndUpdateState.UpgradeConsensusState)
+					suite.Require().Equal(proofUpgradedClient, payload.VerifyUpgradeAndUpdateState.ProofUpgradeClient)
+					suite.Require().Equal(proofUpgradedConsState, payload.VerifyUpgradeAndUpdateState.ProofUpgradeConsensusState)
+
+					// verify other Sudo fields are nil
+					suite.Require().Nil(payload.CheckSubstituteAndUpdateState)
+					suite.Require().Nil(payload.UpdateState)
+					suite.Require().Nil(payload.UpdateStateOnMisbehaviour)
+					suite.Require().Nil(payload.VerifyMembership)
+					suite.Require().Nil(payload.VerifyNonMembership)
+
+					data, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					// set new client state and consensus state
+					store.Set(host.ClientStateKey(), clienttypes.MustMarshalClientState(suite.chainA.App.AppCodec(), upgradedClient))
+					store.Set(host.ConsensusStateKey(upgradedClient.GetLatestHeight()), clienttypes.MustMarshalConsensusState(suite.chainA.App.AppCodec(), upgradedConsState))
+
+					return &wasmvmtypes.Response{Data: data}, types.DefaultGasUsed, nil
+				})
+			},
+			func(err error) {
+				suite.Require().NoError(err)
 
-// 			clientState := path.EndpointA.GetClientState().(*types.ClientState)
-// 			var clientStateData exported.ClientState
-// 			err = suite.chainA.Codec.UnmarshalInterface(clientState.Data, &clientStateData)
-// 			suite.Require().NoError(err)
-// 			tmClientState := clientStateData.(*ibctm.ClientState)
+				// verify new client state and consensus state
+				clientStateBz := suite.store.Get(host.ClientStateKey())
+				suite.Require().NotEmpty(clientStateBz)
+				suite.Require().Equal(upgradedClient, clienttypes.MustUnmarshalClientState(suite.chainA.Codec, clientStateBz))
 
-// 			revisionNumber := clienttypes.ParseChainID(tmClientState.ChainId)
+				consStateBz := suite.store.Get(host.ConsensusStateKey(upgradedClient.GetLatestHeight()))
+				suite.Require().NotEmpty(consStateBz)
+				suite.Require().Equal(upgradedConsState, clienttypes.MustUnmarshalConsensusState(suite.chainA.Codec, consStateBz))
+			},
+		},
+		{
+			"failure: upgraded client state is not wasm client state",
+			func() {
+				// set upgraded client state to solomachine client state
+				upgradedClient = &solomachine.ClientState{}
+			},
+			func(err error) {
+				suite.Require().Error(err)
+				suite.Require().ErrorIs(err, clienttypes.ErrInvalidClient)
 
-// 			newChainID, err = clienttypes.SetRevisionNumber(tmClientState.ChainId, revisionNumber+1)
-// 			suite.Require().NoError(err)
+				// verify client state is unchanged
```

Comment:
Should we also verify that consensus state is unchanged?

Reply:
Done! Made move back to the `expErr` workflow, `expResult` was just more unnecessary repetition.